pkg/object/mqttproxy/storage.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mqttproxy
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/megaease/easegress/pkg/cluster"
+	etcderror "go.etcd.io/etcd/api/v3/v3rpc/rpctypes"
+)
+
+type (
+	storage interface {
+		get(key string) (*string, error)
+		getPrefix(prefix string) (map[string]string, error)
+		put(key, value string) error
+	}
+
+	mockStorage struct {
+		mu    sync.Mutex
```

Comment:
Why not use the `sync.RWMutex`?

Reply:
I think we might need a unit test here to make sure the safety fo concurrent read/write of the map.

Reply:
Thanks a lot for your advices, I will fix them all!